external-crates/move/crates/move-compiler/tests/linter/incorrect_redundant_ref_deref.move

```diff
@@ -0,0 +1,10 @@
+module 0x42::M {
+    struct MyResource has copy, drop{
+        value: u64,
+    }
+
+    public fun test_borrow_deref_ref() {
+        let resource = MyResource { value: 10 };
+        let _ref2 = &*(&resource);
```

Comment:
We will need a lot more tests here
Including but not limited to 
- field borrows, &*&x.f. 
- non local borrows, &*&0 &*&foo()
- method calls `(*&x).foo()` where `fun foo(x: &X)`